skins/oasis/modules/UserPagesHeaderController.class.php

```diff
@@ -71,325 +74,166 @@ public static function getUserName( Title $title, $namespaces, $fallbackToGlobal
-			// "Leave a message" button
-			if ( isset( $this->content_actions['addsection']['href'] ) ) {
-				$actionMenu['action'] = array(
-					'href' => $this->content_actions['addsection']['href'],
-					'text' => wfMsg( 'add_comment' ),
-				);
-
-				$this->actionImage = MenuButtonController::MESSAGE_ICON;
-				$this->actionName = 'leavemessage';
-
-				// different handling for "My talk page"
-				if ( self::isItMe($this->userName ) ) {
-					$actionMenu['action']['text'] = wfMsg( 'edit' );
-					$actionMenu['action']['href'] = $this->content_actions['edit']['href'];
-
-					$this->actionImage = MenuButtonController::EDIT_ICON;
-					$this->actionName = 'edit';
-				}
-			}
-		}
-		else if( defined( 'NS_BLOG_ARTICLE' ) && $namespace == NS_BLOG_ARTICLE ) {
-			// "Create a blog post" button
-			if ( self::isItMe( $this->userName ) ) {
-				$this->actionButton = array(
-					'href' => SpecialPage::getTitleFor( 'CreateBlogPage' )->getLocalUrl(),
-					'text' => wfMsg( 'blog-create-post-label' ),
-				);
-
-				$this->actionImage = MenuButtonController::BLOG_ICON;
-				$this->actionName = 'createblogpost';
-			}
-		}
-
-		// dropdown actions for "Profile" and "Talk page" tabs
-		if ( in_array( $namespace, array( NS_USER, NS_USER_TALK ) ) ) {
-			$actions = array( 'move', 'protect', 'unprotect', 'delete', 'undelete', 'history' );
-
-			// add "edit" item to "Leave a message" button
-			if ( $this->actionName == 'leavemessage' ) {
-				array_unshift( $actions, 'edit' );
-			}
-
-			foreach( $actions as $action ) {
-				if ( isset( $this->content_actions[$action] ) ) {
-					$actionMenu['dropdown'][$action] = $this->content_actions[$action];
-				}
-			}
-		}
-		$this->actionMenu = $actionMenu;
-
-		// don't show stats for user pages with too long titles (RT #68818)
-		if ( mb_strlen( $this->title ) > 35) {
-			$this->stats = false;
-		}
-
-		wfProfileOut( __METHOD__ );
-	}
-
-	/**
-	 * dirty way to get the user url
-	 * returns the full path of the users
-	 **/
-	public static function getUserURL() {
-		global $wgUser;
-
-		$user_name = $wgUser->mName;
-		$userURL = explode( '/', AvatarService::getUrl( $user_name ) );
-		$userURL = $userURL[count( $userURL ) -1];
-		return $userURL;
+		// render tabbed links below masthead
+		$this->tabs = $this->getTabs( $userName );
 	}
 
 	/**
 	 * Render header for blog post
 	 */
-	public function executeBlogPost() {
+	public function blogPost() {
 		wfProfileIn( __METHOD__ );
-		global $wgTitle, $wgLang, $wgOut;
-
 		// remove User_blog:xxx from title
-		$titleParts = explode( '/', $wgTitle->getText() );
+		$titleParts = explode( '/', $this->wg->Title->getText() );
 		array_shift( $titleParts );
 		$this->title = implode( '/', $titleParts );
 
 		// get user name to display in header
-		$this->userName = self::getUserName( $wgTitle, BodyController::getUserPagesNamespaces() );
+		$this->userName = self::getUserName( $this->wg->Title, BodyController::getUserPagesNamespaces() );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)